include/deal.II/base/memory_space.h

```diff
@@ -41,6 +45,12 @@ namespace MemorySpace
 
 } // namespace MemorySpace
 
+#ifdef DEAL_II_USE_KOKKOS_BACKEND
+using HostMemorySpace = ::Kokkos::HostSpace;
+#else
+using HostMemorySpace = MemorySpace::Host;
+#endif
```

Comment:
I'm not a huge fan of this introduction of `HostMeorySpace`. Why not something like
```
namespace MemorySpace
{
  struct Host
#if DEAL_II_USE_KOKKOS_BACKEND
  : public ::Kokkos::HostSpace
#endif
  {};
}
```
? This way, we could neatly keep things in the existing namespace, you'd reduce churn in this patch by half, and where you want to pass something off to Kokkos, you can do that due to the public derivation.

Reply:
I don't think I can make that work with a derived class but I can do
```C++
namespace MemorySpace
{
  struct Host
 {
#if DEAL_II_USE_KOKKOS_BACKEND
  using KokkosSpace = Kokkos::HostSpace
#endif
  };
}
```
Then when I use Kokkos, I can do
```C++
Kokkos::View<int*, MemorySpace::Host::KokkosSpace> a('a');
```

The reason I introduced `HostMemorySpace` is so that later, it is easy to deprecate and then remove `MemorySpace::Host`. Then once it's done we could use directly Kokkos memory spaces instead of hiding them and we could also remove `HostMemorySpace` (or we could keep it since it's a just an alias it doesn't really matter).

Reply:
Yes, the `using` declaration has my vote.

You could also do
```
namespace MemorySpace
{
#if DEAL_II_USE_KOKKOS_BACKEND
  using Host = ::Kokkos::HostSpace;
#else
  struct Host {}; // as before
}
```